crates/iceberg-catalog/src/implementations/postgres/tabular/table.rs

```diff
@@ -222,41 +145,45 @@ pub(crate) async fn create_table(
             .build()
     })?;
 
+    let tabular_id = create_tabular(
+        CreateTabular {
+            id: table_id.into_uuid(),
+            name,
+            namespace_id: namespace_id.into_uuid(),
+            typ: TabularType::Table,
+            metadata_location: metadata_location.map(std::string::String::as_str),
+            location: location.as_str(),
+        },
+        &mut *transaction,
+    )
+    .await?;
+
     // ToDo: Should we keep the old table_id?
     let _update_result = sqlx::query!(
         r#"
-        INSERT INTO "table" (table_id, namespace_id, "table_name", "metadata", "metadata_location", "table_location")
+        INSERT INTO "table" (table_id, "metadata")
         (
-            SELECT $1, $2, $3, $4, $5, $6
-            WHERE EXISTS (
-                SELECT 1
-                FROM warehouse w
-                INNER JOIN namespace n ON w.warehouse_id = n.warehouse_id
-                WHERE n.namespace_id = $2 AND w.status = 'active'
-        ))
-        ON CONFLICT ON CONSTRAINT unique_table_name_per_namespace
-        DO UPDATE SET table_id= $1, "metadata" = $4, "metadata_location" = $5, "table_location" = $6
-        WHERE "table"."metadata_location" IS NULL
+            SELECT $1, $2
+            WHERE EXISTS (SELECT 1
+                FROM active_tables
+                WHERE active_tables.tabular_id = $1))
+        ON CONFLICT ON CONSTRAINT "table_pkey"
+        DO UPDATE SET "metadata" = $2
         RETURNING "table_id"
         "#,
-        table_id.as_uuid(),
-        namespace_id.as_uuid(),
-        name,
+        tabular_id,
         table_metadata_ser,
-        metadata_location,
-        location
     )
     .fetch_one(&mut **transaction)
     .await
-    .map_err(|e| {
-        match &e {
-            sqlx::Error::RowNotFound => ErrorModel::builder()
-                .code(StatusCode::CONFLICT.into())
-                .message("Table already exists in Namespace".to_string())
-                .r#type("TableAlreadyExists".to_string())
-                .build(),
+    .map_err(|e| match &e {
+        sqlx::Error::RowNotFound => ErrorModel::builder()
+            .code(StatusCode::CONFLICT.into())
```

Comment:
Isn't this now an InternalServerError?
If `create_tabular` succeeds, this should also always succeed.